tests/test_coreg.py

```diff
@@ -519,3 +427,324 @@ def test_z_scale_corr(self):
+    assert np.abs(np.nanmedian(diff)) < 0.01
+    # Check that the NMAD is low
+    assert spatial_tools.nmad(diff) < 0.01
+
+    def rotation_matrix(rotation=30):
+        rotation = np.deg2rad(rotation)
+        matrix = np.array([
+            [1, 0, 0, 0],
+            [0, np.cos(rotation), -np.sin(rotation), 0],
+            [0, np.sin(rotation), np.cos(rotation), 0],
+            [0, 0, 0, 1]
+        ])
+        return matrix
+
+    rotation = 4
+    centroid = [np.mean([ref.bounds.left, ref.bounds.right]), np.mean(
+        [ref.bounds.top, ref.bounds.bottom]), ref.data.mean()]
+    rotated_dem = coreg.apply_matrix(
+        ref.data.squeeze(),
+        ref.transform,
+        rotation_matrix(rotation),
+        centroid=centroid
+    )
+    # Make sure that the rotated DEM is way off, but is centered around the same approximate point.
+    assert np.abs(np.nanmedian(rotated_dem - ref.data.data)) < 1
+    assert spatial_tools.nmad(rotated_dem - ref.data.data) > 500
+
+    # Apply a rotation in the opposite direction
+    unrotated_dem = coreg.apply_matrix(
+        rotated_dem,
+        ref.transform,
+        rotation_matrix(-rotation * 0.99),
+        centroid=centroid
+    ) + 4.0  # TODO: Check why the 0.99 rotation and +4 biases were introduced.
+
+    diff = np.asarray(ref.data.squeeze() - unrotated_dem)
+
+    if False:
+        import matplotlib.pyplot as plt
+
+        vmin = 0
+        vmax = 1500
+        extent = (ref.bounds.left, ref.bounds.right, ref.bounds.bottom, ref.bounds.top)
+        plot_params = dict(
+            extent=extent,
+            vmin=vmin,
+            vmax=vmax
+        )
+        plt.figure(figsize=(22, 4), dpi=100)
+        plt.subplot(151)
+        plt.title("Original")
+        plt.imshow(ref.data.squeeze(), **plot_params)
+        plt.xlim(*extent[:2])
+        plt.ylim(*extent[2:])
+        plt.subplot(152)
+        plt.title(f"Rotated {rotation} degrees")
+        plt.imshow(rotated_dem, **plot_params)
+        plt.xlim(*extent[:2])
+        plt.ylim(*extent[2:])
+        plt.subplot(153)
+        plt.title(f"De-rotated {-rotation} degrees")
+        plt.imshow(unrotated_dem, **plot_params)
+        plt.xlim(*extent[:2])
+        plt.ylim(*extent[2:])
+        plt.subplot(154)
+        plt.title("Original vs. de-rotated")
+        plt.imshow(diff, extent=extent, vmin=-10, vmax=10, cmap="coolwarm_r")
+        plt.colorbar()
+        plt.xlim(*extent[:2])
+        plt.ylim(*extent[2:])
+        plt.subplot(155)
+        plt.title("Original vs. de-rotated")
+        plt.hist(diff[np.isfinite(diff)], bins=np.linspace(-10, 10, 100))
+        plt.tight_layout(w_pad=0.05)
+        plt.show()
+
+    # Check that the median is very close to zero
+    assert np.abs(np.nanmedian(diff)) < 0.5
+    # Check that the NMAD is low
+    assert spatial_tools.nmad(diff) < 5
+    print(np.nanmedian(diff), spatial_tools.nmad(diff))
+
+
+def generate_random_field(shape: tuple[int, int], corr_size: int) -> np.ndarray:
```

Comment:
Should it be stored somewhere else, outside of the tests? Such a function could be useful for tutorials for example maybe?

Reply:
Added `xdem/misc.py` for all kinds of small functions like that, and moved the semi-random function there in 366b722